indico/modules/events/abstracts/lists.py

```diff
@@ -115,10 +115,23 @@ def _filter_list_entries(self, query, filters):
 
         if field_filters:
             for contribution_type_id, field_values in field_filters.items():
-                criteria.append(Abstract.field_values.any(db.and_(
+
+                field_value_criteria = db.and_(
                     AbstractFieldValue.contribution_field_id == contribution_type_id,
                     AbstractFieldValue.data.op('#>>')('{}').in_(field_values)
-                )))
+                )
+
+                # Support filtering by 'No selection' in single-choice abstract fields
+                if '' in field_values:
+                    field_value_criteria = db.or_(
+                        field_value_criteria,
+                        db.and_(
+                            AbstractFieldValue.contribution_field_id == contribution_type_id,
+                            AbstractFieldValue.data.op('#>>')('{}').is_(None)
```

Comment:
Does this work if there is literally no data for the field? For example, a field that was created *after* submitting the abstract?

Reply:
I think this is exactly what you commented as well: I think we should handle it by making the query succeed if there is no AbstractFieldValue (ie NOT by updating existing abstracts when adding a field)

Reply:
Yes, that was exactly my question. OK, I'll update the query :+1: